src/google/protobuf/dynamic_message.cc

```diff
@@ -304,10 +304,8 @@ class DynamicMessage final : public Message {
   // class's memory is allocated via the global operator new. Thus, we need to
   // manually call the global operator delete. Calling the destructor is taken
   // care of for us. This makes DynamicMessage compatible with -fsized-delete.
-  // It doesn't work for MSVC though.
-#ifndef _MSC_VER
+  // Working in modern MSVC, may not work in old versions though.
```

Comment:
Rather than removing the guard entirely, can it be updated to check that `_MSC_VER` is new enough when present?